src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Char.cs

```diff
@@ -1718,5 +1718,96 @@ private static bool TryFindFirstMatchedLane(Vector128<ushort> compareResult, ref
             matchedLane = BitOperations.TrailingZeroCount(selectedLanes) >> 3;
             return true;
         }
+
+        public static void Reverse(ref char buf, nuint length)
+        {
+            nint numBytes = (int)length * sizeof(char);
+            int numBytesWritten = 0;
+            ref byte first = ref Unsafe.As<char, byte>(ref buf);
+            ref byte last = ref Unsafe.Add(ref Unsafe.Add(ref first, numBytes), -sizeof(char));
```

Comment:
Semantically, using [Subtract](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.unsafe.subtract?view=net-6.0) would be better.
```suggestion
            ref byte last = ref Unsafe.Subtract(ref Unsafe.Add(ref first, numBytes), sizeof(char));
```